dali/imgcodec/decoders/libtiff/tiff_libtiff.cc

```diff
@@ -234,11 +244,17 @@ void DLL_PUBLIC UnpackBits(size_t nbits, OutputType *out, const void *in, size_t
         bits_in_buffer = buffer_capacity;
       }
     }
-    if (normalize) {
-      double coeff = static_cast<double>((1ull << out_type_bits) - 1) / ((1ull << nbits) - 1);
-      result *= coeff;
+    if (info.is_palette) {
+      out[3*i+0] = ConvertNorm<OutputType>(info.palette.red[result]);
+      out[3*i+1] = ConvertNorm<OutputType>(info.palette.green[result]);
+      out[3*i+2] = ConvertNorm<OutputType>(info.palette.blue[result]);
+    } else {
+      if (normalize) {
+        double coeff = static_cast<double>((1ull << out_type_bits) - 1) / ((1ull << nbits) - 1);
```

Comment:
Is that different from the `ConvertNorm` above?

Reply:
`ConvertNorm` allows you to convert from one type to another, for example from 16-bit integer to 8-bit integer. However, TIFFs can have more exotic bit depths (https://github.com/NVIDIA/DALI/pull/4180), so here we might need to convert from any bitdepth, for example from 19-bit integer – and `ConvertNorm` won't do that for us

Reply:
Oh, sure, now I rember the PR with some reference samples to `DALI_extra` of unusual bit depths. Thanks!